Mathlib/CategoryTheory/Limits/Shapes/Products.lean

```diff
@@ -225,6 +225,24 @@ abbrev Sigma.mapIso {f g : β → C} [HasCoproductsOfShape β C] (p : ∀ b, f b
   colim.mapIso (Discrete.natIso fun X => p X.as)
 #align category_theory.limits.sigma.map_iso CategoryTheory.Limits.Sigma.mapIso
 
+/-- An iterated product is a product over a sigma type. -/
```

Comment:
I think it makes sense to also provide the instance `[∀ i, HasProduct (g i)] [HasProduct fun i => ∏ g i]` -> `HasProduct fun p : Σ i, f i => g p.1 p.2` (and similar for coproducts and biproducts). Here's the product version:
```lean
instance (f : ι → Type _) (g : (i : ι) → (f i) → C)
    [∀ i, HasProduct (g i)] [HasProduct fun i => ∏ g i] :
    HasProduct fun p : Σ i, f i => g p.1 p.2 where
  exists_limit := Nonempty.intro
    { cone := Fan.mk (∏ fun i => ∏ g i) (fun X => Pi.π (fun i => ∏ g i) X.1 ≫ Pi.π (g X.1) X.2)
      isLimit := mkFanLimit _ (fun s => Pi.lift fun b => Pi.lift fun c => s.proj ⟨b, c⟩)
        (by aesop_cat) (by aesop_cat) }
```
Then you can drop the corresponding typeclass assumption on `piPiIso`.